#pragma once

#include <QtWidgets/QMainWindow>
#include "MyView.h"
#include "MyMap.h"
#include "ui_QWalkingMaps.h"
#include<QDebug>
#include<QCompleter>
#include<QMessageBox>

class QWalkingMaps : public QMainWindow
{
    Q_OBJECT
public slots:        
    void SelectFromLocationSignal(int);   //用硖理x衿瘘c信的槽
    void SelectToLocationSignal(int);     //用硖理x窠Kc信的槽
    void ShowError(int);                    //用硖理e`信息K出的槽
    void ShowSameFromOrganizations(const QString&); //用盹@示相同起c名Q的槽
    void ShowSameToOrganizations(const QString&);       //用盹@示相同Kc名Q的槽
    void EnterFromOrganizations();      //用硖理入起c名Q的槽
    void EnterToOrganizations();    //用硖理入Kc名Q的槽

private:
    Ui::QWalkingMapsClass ui;
    QGraphicsScene scene;   // 定义一个场景
    QLabel* FromLabel;
    QLabel* ToLabel;
    map<wstring, int> SameFromOrganizations;
    map<wstring, int> SameToOrganizations;
    int LastSelectFromLocationIndex = -1;
    int LastSelectToLocationIndex = -1;
public:
    QWalkingMaps(QWidget* parent = Q_NULLPTR);
    MyMap *map = new MyMap(scene);
    ~QWalkingMaps();
};
